framework_tvb/tvb/core/entities/file/file_update_scripts/005_update_files.py

```diff
@@ -32,17 +32,669 @@
+        _pop_lengths(root_metadata)
+        root_metadata['connectivity'] = GID_PREFIX + root_metadata['connectivity']
+
+        operation_xml_parameters['data_file'] = input_file
+        operation_xml_parameters['connectivity'] = root_metadata['connectivity']
+
+        _migrate_dataset_metadata(['array_data'], storage_manager)
+        dependent_attributes['connectivity'] = root_metadata['connectivity']
+
+    # TODO: Make sure DatatypeMeasureH5 is converted after a DatatypeMeasure class is created and then remove the return
+    elif class_name == 'DatatypeMeasure':
+        root_metadata['written_by'] = 'tvb.core.entities.file.simulator.datatype_measure_h5.DatatypeMeasureH5'
+        h5_class = DatatypeMeasureH5
+        dependent_attributes['analyzed_datatype'] = root_metadata['analyzed_datatype']
+        return
+
+    elif class_name == 'StimuliRegion':
+        root_metadata['connectivity'] = GID_PREFIX + root_metadata['connectivity']
+        additional_params['weight'] = numpy.asarray(eval(root_metadata['weight']), dtype=numpy.float64)
+        _migrate_stimuli(root_metadata, storage_manager, ['weight'])
+        dependent_attributes['connectivity'] = root_metadata['connectivity']
+
+    elif class_name == 'StimuliSurface':
+        root_metadata['surface'] = GID_PREFIX + root_metadata['surface']
+        additional_params['focal_points_triangles'] = numpy.asarray(eval(root_metadata['focal_points_triangles']),
+                                                                    dtype=numpy.int)
+        _migrate_stimuli(root_metadata, storage_manager, ['focal_points_surface', 'focal_points_triangles'])
+        dependent_attributes['surface'] = root_metadata['surface']
+
+    elif class_name == 'ValueWrapper':
+        root_metadata['data_type'] = root_metadata['data_type'].replace("\"", '')
+        root_metadata['written_by'] = "tvb.adapters.datatypes.h5.mapped_value_h5.ValueWrapperH5"
+        h5_class = ValueWrapperH5
+
+    elif class_name == 'SimulationState':
+        os.remove(input_file)
+        return
+
+    root_metadata['operation_tag'] = ''
+    storage_manager.set_metadata(root_metadata)
+
+    if storage_migrate is False:
+        return
+
+    with h5_class(input_file) as f:
+
+        # Create the corresponding datatype to be stored in db
+        datatype = REGISTRY.get_datatype_for_h5file(f)()
+        f.load_into(datatype)
+        generic_attributes = f.load_generic_attributes()
+        datatype_index = REGISTRY.get_index_for_datatype(datatype.__class__)()
+        datatype_index.create_date = root_metadata['create_date']
+
+        # Get the dependent datatypes
+        for attr_name, attr_value in dependent_attributes.items():
+            dependent_datatype_index = dao.get_datatype_by_gid(attr_value.replace('-', '').replace('urn:uuid:', ''))
+            dependent_datatype = h5.load_from_index(dependent_datatype_index)
+            setattr(datatype, attr_name, dependent_datatype)
```

Comment:
take a look at `h5.load_with_references` to reuse here. It loads the whole datatype from a h5 file together with its dependencies. So the code above would turn into one line that loads the datatype and then the lines that prepare the index.

Reply:
Wish I knew sooner, with this there is no need for the dependent_datatype dict...